tsfc/driver.py

```diff
@@ -267,21 +269,60 @@ def name_multiindex(multiindex, name):
     return builder.construct_kernel(kernel_name, impero_c, parameters["precision"], index_names, quad_rule)
 
 
-def compile_expression_at_points(expression, points, coordinates, interface=None,
-                                 parameters=None, coffee=True):
-    """Compiles a UFL expression to be evaluated at compile-time known
-    reference points.  Useful for interpolating UFL expressions onto
-    function spaces with only point evaluation nodes.
+def apply_mapping(expression, to_element):
```

Comment:
Please document this function call. "Apply mapping" is incredibly generic! Remember that there was a nontrivial discussion about pullbacks and push forwards when this code was written, so it's not totally obvious to a reader what this does.